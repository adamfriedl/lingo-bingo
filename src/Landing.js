// @flow

import React from 'react';
import { Link } from 'react-router-dom';
import { Banner, Text, Button } from 'rebass';
import NavBar from './NavBar';
import Headline from './Headline';

const photo =
  'https://images.unsplash.com/photo-1506701554939-d0aa4c26d79c?auto=format&fit=crop&w=1052&q=60&ixid=dW5zcGxhc2guY29tOzs7Ozs%3D';

const Landing = () => (
  <div>
    <NavBar />
    <Banner
      backgroundImage={photo}
      color="teal3"
      py={[4, 5, 6]}
      mt={-3}
      mx={-3}>
      <Headline text="Lingo Bingo" />
      <Text f={[2, 3, 4]} p={1} center bg="blue9">
        Are Bingo and conference calls your core competencies? <br />
        Sounds like an opportunity for synergy.
      </Text>
      <Link to="/bingocard">
        <Button my={[2, 3]} children="Let's Play!" color="blue9" bg="teal3" />
      </Link>
    </Banner>
  </div>
);

export default Landing;
