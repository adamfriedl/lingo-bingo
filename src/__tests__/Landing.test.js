import React from 'react';
import { shallow } from 'enzyme';
import Landing from '../Landing';

it('renders without crashing', () => {
  const component = shallow(<Landing />);
  expect(component).toMatchSnapshot();
});
