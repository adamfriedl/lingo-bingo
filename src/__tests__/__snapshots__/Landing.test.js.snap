// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavBar />,
        <Styled(Styled(Base))
          backgroundImage="https://images.unsplash.com/photo-1506701554939-d0aa4c26d79c?auto=format&fit=crop&w=1052&q=60&ixid=dW5zcGxhc2guY29tOzs7Ozs%3D"
          color="teal3"
          mt={-3}
          mx={-3}
          py={
                    Array [
                              4,
                              5,
                              6,
                            ]
          }
>
          <Headline
                    text="Lingo Bingo"
          />
          <Styled(Styled(Base))
                    bg="blue9"
                    center={true}
                    f={
                              Array [
                                        2,
                                        3,
                                        4,
                                      ]
                    }
                    p={1}
          >
                    Are Bingo and conference calls your core competencies? 
                    <br />
                    Sounds like an opportunity for synergy.
          </Styled(Styled(Base))>
          <Link
                    replace={false}
                    to="/bingocard"
          >
                    <Styled(Styled(Base))
                              bg="teal3"
                              color="blue9"
                              my={
                                        Array [
                                                  2,
                                                  3,
                                                ]
                              }
                    >
                              Let's Play!
                    </Styled(Styled(Base))>
          </Link>
</Styled(Styled(Base))>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "backgroundImage": "https://images.unsplash.com/photo-1506701554939-d0aa4c26d79c?auto=format&fit=crop&w=1052&q=60&ixid=dW5zcGxhc2guY29tOzs7Ozs%3D",
          "children": Array [
            <Headline
              text="Lingo Bingo"
/>,
            <Styled(Styled(Base))
              bg="blue9"
              center={true}
              f={
                            Array [
                                          2,
                                          3,
                                          4,
                                        ]
              }
              p={1}
>
              Are Bingo and conference calls your core competencies? 
              <br />
              Sounds like an opportunity for synergy.
</Styled(Styled(Base))>,
            <Link
              replace={false}
              to="/bingocard"
>
              <Styled(Styled(Base))
                            bg="teal3"
                            color="blue9"
                            my={
                                          Array [
                                                        2,
                                                        3,
                                                      ]
                            }
              >
                            Let's Play!
              </Styled(Styled(Base))>
</Link>,
          ],
          "color": "teal3",
          "mt": -3,
          "mx": -3,
          "py": Array [
            4,
            5,
            6,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "text": "Lingo Bingo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "bg": "blue9",
              "center": true,
              "children": Array [
                "Are Bingo and conference calls your core competencies? ",
                <br />,
                "Sounds like an opportunity for synergy.",
              ],
              "f": Array [
                2,
                3,
                4,
              ],
              "p": 1,
            },
            "ref": null,
            "rendered": Array [
              "Are Bingo and conference calls your core competencies? ",
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "Sounds like an opportunity for synergy.",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Styled(Base))
                bg="teal3"
                color="blue9"
                my={
                                Array [
                                                2,
                                                3,
                                              ]
                }
>
                Let's Play!
</Styled(Styled(Base))>,
              "replace": false,
              "to": "/bingocard",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "bg": "teal3",
                "children": "Let's Play!",
                "color": "blue9",
                "my": Array [
                  2,
                  3,
                ],
              },
              "ref": null,
              "rendered": "Let's Play!",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavBar />,
          <Styled(Styled(Base))
            backgroundImage="https://images.unsplash.com/photo-1506701554939-d0aa4c26d79c?auto=format&fit=crop&w=1052&q=60&ixid=dW5zcGxhc2guY29tOzs7Ozs%3D"
            color="teal3"
            mt={-3}
            mx={-3}
            py={
                        Array [
                                    4,
                                    5,
                                    6,
                                  ]
            }
>
            <Headline
                        text="Lingo Bingo"
            />
            <Styled(Styled(Base))
                        bg="blue9"
                        center={true}
                        f={
                                    Array [
                                                2,
                                                3,
                                                4,
                                              ]
                        }
                        p={1}
            >
                        Are Bingo and conference calls your core competencies? 
                        <br />
                        Sounds like an opportunity for synergy.
            </Styled(Styled(Base))>
            <Link
                        replace={false}
                        to="/bingocard"
            >
                        <Styled(Styled(Base))
                                    bg="teal3"
                                    color="blue9"
                                    my={
                                                Array [
                                                            2,
                                                            3,
                                                          ]
                                    }
                        >
                                    Let's Play!
                        </Styled(Styled(Base))>
            </Link>
</Styled(Styled(Base))>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "backgroundImage": "https://images.unsplash.com/photo-1506701554939-d0aa4c26d79c?auto=format&fit=crop&w=1052&q=60&ixid=dW5zcGxhc2guY29tOzs7Ozs%3D",
            "children": Array [
              <Headline
                text="Lingo Bingo"
/>,
              <Styled(Styled(Base))
                bg="blue9"
                center={true}
                f={
                                Array [
                                                2,
                                                3,
                                                4,
                                              ]
                }
                p={1}
>
                Are Bingo and conference calls your core competencies? 
                <br />
                Sounds like an opportunity for synergy.
</Styled(Styled(Base))>,
              <Link
                replace={false}
                to="/bingocard"
>
                <Styled(Styled(Base))
                                bg="teal3"
                                color="blue9"
                                my={
                                                Array [
                                                                2,
                                                                3,
                                                              ]
                                }
                >
                                Let's Play!
                </Styled(Styled(Base))>
</Link>,
            ],
            "color": "teal3",
            "mt": -3,
            "mx": -3,
            "py": Array [
              4,
              5,
              6,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "text": "Lingo Bingo",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "bg": "blue9",
                "center": true,
                "children": Array [
                  "Are Bingo and conference calls your core competencies? ",
                  <br />,
                  "Sounds like an opportunity for synergy.",
                ],
                "f": Array [
                  2,
                  3,
                  4,
                ],
                "p": 1,
              },
              "ref": null,
              "rendered": Array [
                "Are Bingo and conference calls your core competencies? ",
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "Sounds like an opportunity for synergy.",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Styled(Base))
                  bg="teal3"
                  color="blue9"
                  my={
                                    Array [
                                                      2,
                                                      3,
                                                    ]
                  }
>
                  Let's Play!
</Styled(Styled(Base))>,
                "replace": false,
                "to": "/bingocard",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "bg": "teal3",
                  "children": "Let's Play!",
                  "color": "blue9",
                  "my": Array [
                    2,
                    3,
                  ],
                },
                "ref": null,
                "rendered": "Let's Play!",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
